website/addons/dataverse/client.py

```diff
@@ -29,7 +29,10 @@ def connect_from_settings(node_settings):
 
 def connect_or_401(host, token):
     try:
-        return _connect(host, token)
+        connection = _connect(host, token)
+        if not connection:
+            raise ConnectionError
```

Comment:
Correct; any error raised here needs to be handled, i.e. a proper error response should be returned to the client.

Also, this logic seems roundabout: you are calling `_connect`, which catches `ConnectionError` and returns `None`. Then here, you are checking `None` and raising a `ConnectionError`.

The purpose of this function is to either return a connection or raise a proper `HTTPError` (which will then be translated into a response to the client); therefore you should probably raise a `HTTPError` rather than a `ConnectionError` here.